### PR TITLE
build.sh: fix hardcode on number of jobs to make

### PR DESCRIPTION
**Pull request by pkalever**
Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

### CODE REVIEW

**Comment by raghavendra-talur** [comment]
+1

File: build.sh [hunk 1]
```diff
@@ -16,5 +27,5 @@ mkdir -p $P/scratch;
 cd $P/scratch;
 rm -rf $P/install;
 $SRC/configure --prefix=$P/install --with-mountutildir=$P/install/sbin --with-initdir=$P/install/etc --localstatedir=/var --enable-bd-xlator=yes --enable-debug --silent
-make install CFLAGS="-g -O0 -Wall -Werror" -j 4 >/dev/null
+make install CFLAGS="-g -O0 -Wall -Werror" -j ${nproc} >/dev/null
```
Comment (kshlm) [inline build.sh]: IIRC, an empty `-j` option should pick the suitable number of parallel jobs to execute.

Comment (jdarcy) [inline build.sh]: According to the man page, -j without an option will not constrain the number of jobs, which I think means it will just spawn as many as it can based on dependencies.  Not sure if that's a "suitable" number for our purposes or not.

Comment (pkalever) [inline build.sh]: I am convinced with the explanation; given by @kshlm @jdarcy 
Got super excited after noticing an hard coded value '4'

Comment (jdarcy) [inline build.sh]: Are you saying that you think we should just use -j without an argument?  That's fine - the platform check is still useful for other things - but it wasn't clear whether that's what you meant.

Comment (pkalever) [inline build.sh]: @kshlm @jdarcy @raghavendra-talur 

After my investigation on make jobs this is the summary:

 "make -j" where if you don't give it a parameter it uses a crazy default (basically, run as many as jobs as it can in parallel - even a hundred).

Actually, number of threads should be proportional to the hardware threads (may be 1.5x),  'make -j' can't limit to hardware_threads X nproc [ X 1.5 ] and goes far beyond the limit.

In fact I have seen VM hung followed by OOM kill in my local VM while compiling linux kernel with 'make -j'.

Basically In the VMs "Thread(s) per core:   1" (mostly), hence its a best practice to limit it to nprocs meaning use 'make -j${nproc}'.

So I still think this patch is valid  :)

**Comment by nigelbabu** [comment]
@pkalever This is still relevant, right? Shall I merge it in?

**Comment by nigelbabu** [comment]
This should probably be closed in favor of #18. Thoughts?

**Comment by pkalever** [comment]
@nigelbabu please push it in #18
